java/src/main/c++/jni_spark_vw.cc

```diff
@@ -505,10 +596,145 @@ JNIEXPORT jobject JNICALL Java_org_vowpalwabbit_spark_VowpalWabbitExample_predic
     // as this is not a ring-based example it is not free'd
     VW::LEARNER::as_singleline(all->l)->finish_example(*all, *ex);
 
-    return Java_org_vowpalwabbit_spark_VowpalWabbitExample_getPrediction(env, exampleObj);
+    return getJavaPrediction(env, all, ex);
+  }
+  catch (...)
+  {
+    rethrow_cpp_exception_as_java_exception(env);
+    // return null
+    return 0;
+  }
+}
+
+JNIEXPORT jstring JNICALL Java_org_vowpalwabbit_spark_VowpalWabbitExample_toString(JNIEnv* env, jobject exampleObj)
+{
+  INIT_VARS
+
+  try
+  {
+    std::ostringstream ostr;
+
+    ostr << "VowpalWabbitExample(label=";
+    auto lp = all->p->lp;
+
+    if (!memcmp(&lp, &simple_label, sizeof(lp)))
+    {
+      label_data* ld = (label_data*)&ex->l;
+      ostr << "simple " << ld->label << ":" << ld->weight << ":" << ld->initial;
+    }
+    else if (!memcmp(&lp, &CB::cb_label, sizeof(lp)))
+    {
+      CB::label* ld = (CB::label*)&ex->l;
+      ostr << "CB " << ld->costs.size();
+
+      if (ld->costs.size() > 0)
+      {
+        ostr << " ";
+
+        CB::cb_class& f = ld->costs[0];
+
+        // ignore validation of action which is uniform_hash("shared")

```

Comment:
what does this comment mean? where are we ignoring it exactly

Reply:
Updated comment. Means we don't check f.action